python/lsst/ts/m2/csc.py

```diff
@@ -330,19 +332,43 @@ async def _telemetry_loop(self):
 
         self.log.debug("Starting telemetry loop from component.")
 
+        messages_consumed = 0
+        messages_consumed_log_timer = asyncio.create_task(
+            asyncio.sleep(self.heartbeat_interval)
+        )
         while self._run_loops:
 
-            if self.model.are_clients_connected() and (
-                not self.model.client_telemetry.queue.empty()
-            ):
-                message = self.model.client_telemetry.queue.get_nowait()
+            if self.model.are_clients_connected():
+                message = (
+                    self.model.client_telemetry.queue.get_nowait()
+                    if not self.model.client_telemetry.queue.empty()
+                    else await self.model.client_telemetry.queue.get()
```

Comment:
`await self.model.client_telemetry.get()` is a blocking call. Therefore it will have the similar problem we discussed in the yesterday. The main thing is that once it blocks in the edge case, how could we close the application actively? Or maybe we do not need to consider this because this is in CSC level?